util/grub.d/00_menu_auto_hide.in

```diff
@@ -1,15 +1,20 @@
 #! /bin/sh
 
-# Disable / skip generating menu-auto-hide config parts on serial terminals
-for x in ${GRUB_TERMINAL_INPUT} ${GRUB_TERMINAL_OUTPUT}; do
-  case "$x" in
-    serial*)
-      exit 0
-      ;;
-  esac
-done
-
 cat << EOF
+# Boot Counting
+if [ "\${boot_counter}" ]; then
+  if [ "\${boot_success}" = "0"]; then
+    if  [ "\${boot_counter}" = "0" -o "\${boot_counter}" = "-1" ]; then
+      set default=1
+      set boot_counter=-1
+      save_env boot_counter
+    else
+      set boot_counter=$((\${boot_counter}-1))
+      save_env boot_counter
```

Comment:
Minor: we can just factor out the `save_env boot_counter` outside the if-statement, right?

Reply:
@jlebon agreed.